File: app/src/main/java/com/alorma/timelineview/app/EventsAdapter.java
/*
package com.alorma.timelineview.app;

import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import com.alorma.timeline.TimelineView;
import java.util.List;

class EventsAdapter extends RecyclerView.Adapter<EventsAdapter.ViewHolderItem> {

  private final LayoutInflater inflater;
  private final List<Event> events;

  EventsAdapter(LayoutInflater inflater, List<Event> events) {
    this.inflater = inflater;
    this.events = events;
  }

  @Override
  public ViewHolderItem onCreateViewHolder(ViewGroup parent, int viewType) {
    return new ViewHolderItem(inflater.inflate(R.layout.item_main, parent, false));
  }

  @Override
  public void onBindViewHolder(ViewHolderItem holder, int position) {
    Event event = events.get(position);

    holder.text.setText(event.getName());
    holder.timeline.setTimelineType(event.getType());
    holder.timeline.setTimelineAlignment(event.getAlignment());
  }

  @Override
  public int getItemCount() {
    return events.size();
  }

  static class ViewHolderItem extends RecyclerView.ViewHolder {
    TextView text;
    TimelineView timeline;

    ViewHolderItem(View itemView) {
      super(itemView);

      text = (TextView) itemView.findViewById(R.id.textView);
      timeline = (TimelineView) itemView.findViewById(R.id.timeline);
    }
  }
}
*/

File: app/src/main/java/com/alorma/timelineview/app/EventsAdapter.kt
package com.alorma.timelineview.app

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.alorma.timeline.property.LineStyle
import com.alorma.timeline.property.PointStyle
import kotlinx.android.synthetic.main.item_main.view.*

class EventsAdapter : ListAdapter<Event, EventsAdapter.ViewHolderItem>(DIFF_CALLBACK) {

    var lineStyle: SampleLineStyle = SampleLineStyle.MIXED
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    var lineColor: SampleLineColor = SampleLineColor.MIXED
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    var pointStyle: SamplePointStyle = SamplePointStyle.MIXED
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolderItem {
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.item_main, parent, false)
        return ViewHolderItem(view)
    }

    override fun onBindViewHolder(holder: ViewHolderItem, position: Int) {
        holder.bind(getItem(position),
                lineStyle, lineColor,
                pointStyle)
    }

    class ViewHolderItem(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(event: Event,
                 lineStyle: SampleLineStyle,
                 lineColor: SampleLineColor,
                 pointStyle: SamplePointStyle) {
            itemView.textView.text = event.name

            configureTimeline(lineStyle, lineColor, event, pointStyle)
        }

        private fun configureTimeline(lineStyle: SampleLineStyle,
                                      lineColor: SampleLineColor,
                                      event: Event,
                                      pointStyle: SamplePointStyle) {
            itemView.timeline.configureLine {
                val lineSty = when (lineStyle) {
                    is SampleLineStyle.LINE -> LineStyle.LINEAR
                    is SampleLineStyle.DASHED -> LineStyle.DASHED
                    is SampleLineStyle.MIXED -> if ((adapterPosition % 2) == 0) {
                        LineStyle.LINEAR
                    } else {
                        LineStyle.DASHED
                    }
                }
                setLineStyle(lineSty)

                val color = when (lineColor) {
                    SampleLineColor.RED -> Color.RED
                    SampleLineColor.GREEN -> Color.GREEN
                    SampleLineColor.MIXED -> if ((adapterPosition % 2) == 0) {
                        Color.RED
                    } else {
                        Color.GREEN
                    }
                }

                setLineColor(color)

                event.lineWidth?.let { width -> setLineWidth(width) }
            }

            itemView.timeline.configurePoint {
                val pointSty = when (pointStyle) {
                    is SamplePointStyle.CIRCLE -> PointStyle.CIRCLE
                    is SamplePointStyle.SQUARE -> PointStyle.SQUARE
                    is SamplePointStyle.MIXED -> if ((adapterPosition % 2) == 0) {
                        PointStyle.CIRCLE
                    } else {
                        PointStyle.SQUARE
                    }
                }

                setPointStyle(pointSty)
            }
        }
    }

    companion object {
        private val DIFF_CALLBACK = object : DiffUtil.ItemCallback<Event>() {
            override fun areItemsTheSame(oldItem: Event, newItem: Event): Boolean =
                    oldItem == newItem

            override fun areContentsTheSame(oldItem: Event, newItem: Event): Boolean =
                    oldItem.name == newItem.name
        }
    }
}